### ISSUE TITLE
HSTP

### ISSUE THREAD

**Issue by ferrine** [pymc-devs/pymc#6920]
### Description

After some research, seems like if you replace `Normal` distribution with `StudentT` distribution and get HSTP

Here is a related paper as well
https://proceedings.mlr.press/v202/sellier23a.html

https://github.com/pymc-devs/pymc/blob/bc4b51dcaea4e0e9fdf8268672bb996a18628944/pymc/gp/hsgp_approx.py#L322-L331

Turning into

```python
        if self._parameterization == "noncentered":
            self._beta = pm.StudentT(
                f"{name}_hsgp_coeffs_", self.nu, size=self._m_star - int(self._drop_first)
            )
            self._sqrt_psd = sqrt_psd
            f = self.mean_func(X) + phi @ (self._beta * self._sqrt_psd)


        elif self._parameterization == "centered":
            self._beta = pm.StudentT(f"{name}_hsgp_coeffs_", self.nu, sigma=sqrt_psd)
            f = self.mean_func(X) + phi @ self._beta
```

Alternatively, you can use Fisher transfrom or any other inverse normalizing transformations to "pull out" `nu`

https://www.jstor.org/stable/2334301

**Comment by bwengals** [comment]
That's really cool, this should definitely be added.  Probably could just put an additional kwarg `nu=None` on the HSGP class and fix the code like you did.  

I don't follow the second paper though, about transforming a t distributed RV to a normal.  What do you mean to pull out `nu`?

**Comment by ferrine** [comment]
> What do you mean to pull out nu?
It specifies non-centered parameterization (non-exact) for Student T by transforming normal distribution to StudentT

**Comment by bwengals** [comment]
ah I see now.  I'm not sure which would work better.  It's an approximation right?  The first way you did it, `pm.Normal` -> `pm.StudentT` is exact and looks simpler to me

**Comment by ferrine** [comment]
Yes, I tried exact one and it works like charm

**Comment by ferrine** [comment]
Should it come with HSGP name or HSTP subclass? GP is Gaussian process, while T Process is different

**Comment by bwengals** [comment]
True... definitely less code to have `nu` be an optional kwarg that makes it a TP.  But you're right, the code would say "GP" when it's really a t-process.  

I kinda prefer the kwarg option?  Though there already is a `TP` class

**Comment by ferrine** [comment]
I prefer the explicit way to create a subclass

**Comment by bwengals** [comment]
From the user perspective I agree.  For implementing it'd make for a lot of code dup, unless I'm missing something clever.  Maybe have kwargs for HSGP, and also an HSTP class that calls it?

**Comment by fonnesbeck** [comment]
If we want to generalize, we could rename the class `HSP` or `HSSP` (stochastic process), and then specify the type of process as a keyword arg.
